### ISSUE TITLE
[BUG] 302 download responses are not followed

### ISSUE THREAD

**Issue by opatut** [onthegomap/flatmap#21]
**Describe the bug**

When downloading area `.osm.pbf` files, a 302 redirect is not followed, instead, an exception is raised.

**To Reproduce**

```
java -Xmx1g -jar flatmap.jar --download --area=germany
```
**Stack trace**

<details>
```
Exception in thread "main" java.lang.IllegalStateException: Error getting size of https://download.geofabrik.de/europe/germany-latest.osm.pbf
	at com.onthegomap.flatmap.util.Downloader.run(Downloader.java:178)
	at com.onthegomap.flatmap.FlatmapRunner.download(FlatmapRunner.java:574)
	at com.onthegomap.flatmap.FlatmapRunner.run(FlatmapRunner.java:475)
	at com.onthegomap.flatmap.basemap.BasemapMain.run(BasemapMain.java:50)
	at com.onthegomap.flatmap.basemap.BasemapMain.main(BasemapMain.java:14)
	at com.onthegomap.flatmap.Main.main(Main.java:50)
Caused by: java.util.concurrent.ExecutionException: java.lang.IllegalStateException: Bad response: 302
	at java.base/java.util.concurrent.CompletableFuture.reportGet(CompletableFuture.java:396)
	at java.base/java.util.concurrent.CompletableFuture.get(CompletableFuture.java:2096)
	at com.onthegomap.flatmap.util.Downloader.run(Downloader.java:175)
	... 5 more
Caused by: java.lang.IllegalStateException: Bad response: 302
	at com.onthegomap.flatmap.util.Downloader.assertOK(Downloader.java:94)
	at com.onthegomap.flatmap.util.Downloader.lambda$httpHead$7(Downloader.java:232)
	at java.net.http/jdk.internal.net.http.Http1Exchange.readBodyAsync(Http1Exchange.java:373)
	at java.net.http/jdk.internal.net.http.Exchange.readBodyAsync(Exchange.java:185)
	at java.net.http/jdk.internal.net.http.MultiExchange.lambda$responseAsync0$4(MultiExchange.java:353)
	at java.base/java.util.concurrent.CompletableFuture$UniCompose.tryFire(CompletableFuture.java:1150)
	at java.base/java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:510)
	at java.base/java.util.concurrent.CompletableFuture.postFire(CompletableFuture.java:614)
	at java.base/java.util.concurrent.CompletableFuture$UniApply.tryFire(CompletableFuture.java:653)
	at java.base/java.util.concurrent.CompletableFuture$Completion.run(CompletableFuture.java:482)
	at java.net.http/jdk.internal.net.http.HttpClientImpl$DelegatingExecutor.execute(HttpClientImpl.java:157)
	at java.base/java.util.concurrent.CompletableFuture$UniCompletion.claim(CompletableFuture.java:572)
	at java.base/java.util.concurrent.CompletableFuture$UniApply.tryFire(CompletableFuture.java:642)
	at java.base/java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:510)
	at java.base/java.util.concurrent.CompletableFuture.complete(CompletableFuture.java:2147)
	at java.net.http/jdk.internal.net.http.Http1Response$HeadersReader.handle(Http1Response.java:695)
	at java.net.http/jdk.internal.net.http.Http1Response$HeadersReader.handle(Http1Response.java:621)
	at java.net.http/jdk.internal.net.http.Http1Response$Receiver.accept(Http1Response.java:612)
	at java.net.http/jdk.internal.net.http.Http1Response$HeadersReader.tryAsyncReceive(Http1Response.java:668)
	at java.net.http/jdk.internal.net.http.Http1AsyncReceiver.flush(Http1AsyncReceiver.java:233)
	at java.net.http/jdk.internal.net.http.common.SequentialScheduler$LockingRestartableTask.run(SequentialScheduler.java:205)
	at java.net.http/jdk.internal.net.http.common.SequentialScheduler$CompleteRestartableTask.run(SequentialScheduler.java:149)
	at java.net.http/jdk.internal.net.http.common.SequentialScheduler$SchedulableTask.run(SequentialScheduler.java:230)
	at java.net.http/jdk.internal.net.http.HttpClientImpl$DelegatingExecutor.execute(HttpClientImpl.java:157)
	at java.net.http/jdk.internal.net.http.common.SequentialScheduler.runOrSchedule(SequentialScheduler.java:305)
	at java.net.http/jdk.internal.net.http.common.SequentialScheduler.runOrSchedule(SequentialScheduler.java:274)
	at java.net.http/jdk.internal.net.http.Http1AsyncReceiver.asyncReceive(Http1AsyncReceiver.java:464)
	at java.net.http/jdk.internal.net.http.Http1AsyncReceiver$Http1TubeSubscriber.onNext(Http1AsyncReceiver.java:584)
	at java.net.http/jdk.internal.net.http.Http1AsyncReceiver$Http1TubeSubscriber.onNext(Http1AsyncReceiver.java:541)
	at java.net.http/jdk.internal.net.http.common.SSLTube$DelegateWrapper.onNext(SSLTube.java:210)
	at java.net.http/jdk.internal.net.http.common.SSLTube$SSLSubscriberWrapper.onNext(SSLTube.java:492)
	at java.net.http/jdk.internal.net.http.common.SSLTube$SSLSubscriberWrapper.onNext(SSLTube.java:295)
	at java.net.http/jdk.internal.net.http.common.SubscriberWrapper$DownstreamPusher.run1(SubscriberWrapper.java:316)
	at java.net.http/jdk.internal.net.http.common.SubscriberWrapper$DownstreamPusher.run(SubscriberWrapper.java:259)
	at java.net.http/jdk.internal.net.http.common.SequentialScheduler$LockingRestartableTask.run(SequentialScheduler.java:205)
	at java.net.http/jdk.internal.net.http.common.SequentialScheduler$CompleteRestartableTask.run(SequentialScheduler.java:149)
	at java.net.http/jdk.internal.net.http.common.SequentialScheduler$SchedulableTask.run(SequentialScheduler.java:230)
	at java.net.http/jdk.internal.net.http.common.SequentialScheduler.runOrSchedule(SequentialScheduler.java:303)
	at java.net.http/jdk.internal.net.http.common.SequentialScheduler.runOrSchedule(SequentialScheduler.java:256)
	at java.net.http/jdk.internal.net.http.common.SubscriberWrapper.outgoing(SubscriberWrapper.java:232)
	at java.net.http/jdk.internal.net.http.common.SubscriberWrapper.outgoing(SubscriberWrapper.java:198)
	at java.net.http/jdk.internal.net.http.common.SSLFlowDelegate$Reader.processData(SSLFlowDelegate.java:444)
	at java.net.http/jdk.internal.net.http.common.SSLFlowDelegate$Reader$ReaderDownstreamPusher.run(SSLFlowDelegate.java:268)
	at java.net.http/jdk.internal.net.http.common.SequentialScheduler$LockingRestartableTask.run(SequentialScheduler.java:205)
	at java.net.http/jdk.internal.net.http.common.SequentialScheduler$CompleteRestartableTask.run(SequentialScheduler.java:149)
	at java.net.http/jdk.internal.net.http.common.SequentialScheduler$SchedulableTask.run(SequentialScheduler.java:230)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at java.base/java.lang.Thread.run(Thread.java:833)
```

</details>

**Expected behavior**

The 302 redirect should be followed and the download should succeed.

**Environment (please complete the following information):**

```
Linux hostname 5.15.5-arch1-1 #1 SMP PREEMPT Thu, 25 Nov 2021 22:09:33 +0000 x86_64 GNU/Linux
openjdk version "17.0.1" 2021-10-19
OpenJDK Runtime Environment (build 17.0.1+12)
OpenJDK 64-Bit Server VM (build 17.0.1+12, mixed mode)
```

**Additional context**

As you can see, `download.geofabrik.de` returns a 302 redirect:

```
$ curl -v https://download.geofabrik.de/europe/germany-latest.osm.pbf 2>&1 | grep -E '< HTTP|Location'
< HTTP/1.1 302 Found
< Location: http://ftp5.gwdg.de/pub/misc/openstreetmap/download.geofabrik.de/germany-latest.osm.pbf
```

**Comment by opatut** [comment]
It seems like not all files on geofabrik are getting the redirect, so monaco works for me. That is probably some CDN logic.

If you encounter this issue, of course, downloading the file yourself and placing it inside `./data/sources/` works wonderfully:

```
wget https://download.geofabrik.de/europe/germany-latest.osm.pbf -O data/sources/germany.osm.pbf
java -Xmx1g -jar flatmap.jar --area=germany
```

**Comment by msbarry** [comment]
Thanks for pointing this out @opatut. I implemented a fix in #23 and tested it out on Germany.  It ran out of memory with `-Xmx1g` but finished fine in about 10 minutes with `-Xmx2g` (2021 macbook m1 pro).
